src/main/java/uk/gov/ons/ctp/response/collection/exercise/service/CollectionExerciseService.java

```diff
@@ -106,6 +106,39 @@ public CollectionExerciseService(
     return this.collectRepo.findBySurveyId(UUID.fromString(survey.getId()));
   }
 
+  /**
+   * Find a list of collection exercises associated with a list of surveys from the Collection Exercise Service
+   *
+   * @param surveyIds the survey UUIDS for which to find collection exercises
+   * @return the associated collection exercises.
+   */
+  public HashMap<UUID, List<CollectionExercise>> findCollectionExercisesForSurveys(List<UUID> surveyIds) {
+
+    List<CollectionExercise> collexList = this.collectRepo.findBySurveyIdInOrderBySurveyId(surveyIds);
+
+    return this.collexListToDict(collexList);
+  }
+
+  /**
+   * Walks a list of collection exercises and splits them into a dictionary with key of survey id
```

Comment:
🎨 dictionary should probably be HashMap as we're not in a python repo (unfortunately :P)

Reply:
Changing